app/utils/serializers.py

```diff
@@ -0,0 +1,213 @@
+import typing as tp
+import sqlalchemy
+import decimal
+import datetime
+
+from flask_sqlalchemy import DefaultMeta
+
+from pydantic import BaseModel, constr, root_validator
+from pydantic.fields import ModelField
+
+SQLALCHEMY_TYPE_TRANSLATE = {
+    sqlalchemy.sql.sqltypes.BigInteger: int,
+    sqlalchemy.sql.sqltypes.Unicode: str,
+    sqlalchemy.sql.sqltypes.Integer: int,
+    sqlalchemy.sql.sqltypes.String: str,
+    sqlalchemy.sql.sqltypes.Float: float,
+    sqlalchemy.sql.sqltypes.Numeric: decimal.Decimal,
+    sqlalchemy.sql.sqltypes.DateTime: tp.Optional[datetime.datetime],
+    sqlalchemy.sql.sqltypes.LargeBinary: bytes,
+    sqlalchemy.sql.sqltypes.Boolean: bool,
+    sqlalchemy.sql.sqltypes.BOOLEAN: bool,
+    sqlalchemy.sql.sqltypes.Date: tp.Optional[datetime.date],
+    sqlalchemy.sql.sqltypes.Time: tp.Optional[datetime.time],
+    sqlalchemy.sql.sqltypes.Interval: tp.Optional[datetime.timedelta],
+    sqlalchemy.sql.sqltypes.ARRAY: list,
+    sqlalchemy.sql.sqltypes.JSON: dict
+}
+
+serializer_data_type = tp.Dict[str, tp.Union[str, int, bool]]
+
+
+class BaseSerializer(BaseModel):
+    """
+    Base serializer
+    """
+
+    remove_fields: tp.List[str] = ['read_only_fields', 'multiple_data', 'many',
+                                   'write_only_fields', 'method', 'remove_fields', '_sa_instance_state',
+                                   'data', 'is_annotate', 'exclude_fields']
+    read_only_fields = []  # type: ignore
+    exclude_fields = []  # type: ignore
+    write_only_fields = []  # type: ignore
+
+    def update_remove_fields(self, fields: tp.List[str]) -> None:
+        """
+        Update remove fields
+        Args:
+            fields: fields which will be append to remove fields
+        """
+
+        for field in fields:
+            self.remove_fields.append(field)
+
+    def validate_data_before_create(self) -> serializer_data_type:
+        """
+        Validate data before create
+        Return:
+            dict with data
+        """
+
+        data = self.__dict__.copy()
+        for key in self.__dict__:
+            is_key_in = bool(
+                key in self.read_only_fields or key in self.remove_fields or key in self.exclude_fields
+            )
+            if data[key] == '' or data[key] is None or is_key_in:
+                del data[key]
+        return data
+
+    def validate_data_before_get(self) -> serializer_data_type:
+        data = self.__dict__.copy()
+        for key in self.__dict__:
+            if key in self.remove_fields or key in self.write_only_fields:
+                del data[key]
+        return data
+
+
+class BaseModelSerializer(BaseSerializer):
+    """
+    Base Model serializer
+    """
+
+    def __init__(self, *args: tp.Any, **kwargs: tp.Any) -> None:
```

Comment:
are you sure such a complex code is needed? can you use https://pydantic-docs.helpmanual.io/usage/models/#orm-mode-aka-arbitrary-class-instances ?